### PR TITLE
feat: [CDS-76504]: Nav 2 latest changes

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/RouteDefinitionsV2.ts [hunk 1]
```diff
@@ -140,46 +147,195 @@ const routes = {
...
-  toChaosDashboards: withModeModuleAndScopePrefix<ModulePathParams>(() => '/chaos-dashboards'),
-  toChaosEnvironments: withModeModuleAndScopePrefix<ModulePathParams>(() => '/environments'),
-  toChaosEnvironmentDetails: withModeModuleAndScopePrefix<ModulePathParams & { identifier: string }>(
-    params => `/environments/${params?.identifier}`
-  ),
-  toChaosSecurityGovernance: withModeModuleAndScopePrefix<ModulePathParams>(() => '/chaos-guard'),
-  toChaosSecurityGovernanceRuleDetails: withModeModuleAndScopePrefix<ModulePathParams & { identifier: string }>(
-    params => `/chaos-guard/rules/${params?.identifier}`
-  ),
-  toChaosImageRegistry: withModeModuleAndScopePrefix<ModulePathParams>(() => '/image-registry'),
```
Comment (SahilKr24) [inline src/modules/10-common/RouteDefinitionsV2.ts:174]: why are we deleting the chaos routes?

File: src/framework/AppStore/AppStoreContext.tsx [hunk 2]
```diff
@@ -116,14 +117,26 @@ export function AppStoreProvider({ children }: PropsWithChildren<unknown>): Reac
   const { showError } = useToaster()
   const history = useHistory()
 
+  const match = matchPath<ModePathProps & ModulePathParams>(location.pathname, {
+    path: MODE_PATH
+  })
+  const modeFromPath = match?.params?.mode
+
+  const moduleMatch = matchPath<ModulePathParams>(location.pathname, {
```
Comment (avinashmadhwani02) [inline src/framework/AppStore/AppStoreContext.tsx:125]: we don't need this matchPath, there's a method present in routeUtils (getParams), we can use that

File: src/modules/10-common/RouteDefinitionsV2.ts [hunk 3]
```diff
@@ -110,14 +107,24 @@ const routes = {
...
+
   //pipeline routes
   toDeployments: withModeModuleAndScopePrefix(() => '/deployments'),
   toPipelines: withModeModuleAndScopePrefix(() => '/pipelines'),
   toPipelineStudio: withModeModuleAndScopePrefix<PipelinePathProps>(
     params => `/pipelines/${params?.pipelineIdentifier}/pipeline-studio`
   ),
   // cd routes
-
+  toModule: ({
```
Comment (avinashmadhwani02) [inline src/modules/10-common/RouteDefinitionsV2.ts:119]: we don't need this method. use toMode instead

File: src/modules/10-common/RouteDefinitionsV2.ts [hunk 4]
```diff
@@ -1,52 +1,64 @@
...
   Partial<ProjectPathProps> &
   Partial<ModePathProps> &
   Partial<ModulePathParams>
 
-export function withModeModuleAndScopePrefix<T>(fn: (args?: T) => string) {
+export function withModeModuleAndScopePrefix<T>(fn: (args?: T) => string, includeCurrentModuleAndScope?: boolean) {
   return (params?: T & PathProps): string => {
-    const match = matchPath<ModePathProps & AccountPathProps & ModulePathParams>(location.pathname, {
-      path: `${window.harnessNameSpace || ''}/ng/account/:accountId/:mode`
+    const match = matchPath<ModePathProps & AccountPathProps & ModulePathParams & ProjectPathProps>(location.pathname, {
```
Comment (avinashmadhwani02) [inline src/modules/10-common/RouteDefinitionsV2.ts:50]: matchPath can be removed

File: src/modules/10-common/RouteDefinitionsV2.ts (hunk 1 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/RouteDefinitionsV2.ts:174]: Got removed while fixing conflicts. adding it back

File: src/modules/10-common/navigation/SideNavs/ModuleSideNav/ModuleSideNav.module.scss
Comment (avinashmadhwani02) [inline src/modules/10-common/navigation/SideNavs/ModuleSideNav/ModuleSideNav.module.scss:1]: Remove this file

File: src/modules/10-common/RouteDefinitionsV2.ts (hunk 1 above)
Comment (SahilKr24) [reply src/modules/10-common/RouteDefinitionsV2.ts:174]: Thanks, please check other chaos files as well, didn't see those as well.

File: src/modules/70-pipeline/pages/execution-list/ExecutionListEmpty/ExecutionListEmpty.module.scss [hunk 5]
```diff
@@ -28,7 +28,7 @@
 }
 
 .overviewNoExecutions {
-  z-index: 2 !important;
+  z-index: 13 !important;
```
Comment (avinashmadhwani02) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListEmpty/ExecutionListEmpty.module.scss:31]: regression risk

File: src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroups.tsx [hunk 6]
```diff
@@ -229,7 +233,7 @@ export default function EnvironmentGroupsPage(): React.ReactElement {
             <HarnessDocTooltip tooltipId={'ff_env_group_heading'} useStandAlone />
           </Heading>
         }
-        toolbar={<EnvironmentTabs />}
+        toolbar={<EnvironmentTabs calledFromSettingsPage={calledFromSettingsPage} />}
```
Comment (avinashmadhwani02) [inline src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroups.tsx:236]: change name to isParentSettingsPage

**Pull request merged**